### PR TITLE
Updated & added docs wrt hall & new talk

### PR DESCRIPTION
**Pull request by Fang-**
This includes a few things:

* Updated `messaging.md` to inform users about all the functionality the new cli talk client has to offer,
* A hall architecture document, outlining the structures, interface and inner mechanics of our generic messaging bus,
* A document explaining the general structure and reasoning behind the new gall model.

Aside from `messaging.md`, I don't think this is quite ready to be merged yet, but I'm making this PR to get some discussion going on what needs to be improved here to make it so.  
Both the hall and gall documents haven't changed all that much from their fora post versions, aside from bringing them up to speed with recent changes to Hall. (The gall document also contains some questions and criticisms I originally included, that probably has to go somewhere else.)

"Depends" on urbit/arvo#452.

### CODE REVIEW

File: docs/using/messaging.md [hunk 1]
```diff
@@ -11,60 +11,57 @@ title: Messaging (:talk)
...
 
 Let's join `/urbit-meta`:
 
 Use `ctrl-x` to switch from `:dojo` to `:talk`.
 
-If you're a planet:
-
-    ~your-urbit:talk> ;join /urbit-meta
-
-If you're a comet:
```
Comment (keatondunsford) [inline docs/using/messaging.md]: I'm assuming `;join /urbit-meta` works for comets now on new Talk?

Comment (Fang-) [reply docs/using/messaging.md]: We're going to want to host urbit-meta on ~zod (for federation), so yeah.

**Comment by keatondunsford** [comment]
It might make sense to wait to merge this until we merge new Talk/Hall into master (or when we breach), but I also don't really see any reason not to get this stuff into the docs now. @Fang- thoughts?

**Comment by keatondunsford** [comment]
I guess we should hold off because of the changes to `/docs/using/messaging.md`.

**Comment by Fang-** [comment]
Yeah I'd hold off until everything's gotten the a-okay and new talk gets pulled into master. (And obviously don't put this live before the breach.)

Also the Hall architecture and interface documents aren't really linked from anywhere yet. Taking suggestions on what a good place might be.

**Comment by keatondunsford** [comment]
My [developer docs work](https://github.com/urbit/docs/tree/developer) generally is on pause for a little while because @galenwp wants to take some time to think about redesigning it (probably with help from @jyng and @c-johnson). Once I pick all this back up, I'll think about a good place for this stuff to go. Just keep writing stuff!

**Pull request merged**